File: backend/Models/Project.cs
namespace backend.Models;

public class Project
{
    public Guid Id{get;set;}
    public Guid UserId{get;set;}
    public string ProjectTitle{get;set;}=string.Empty;
    public string Description{get;set;}=string.Empty;
    public DateTime StartDate{get;set;}
    public DateTime EndDate{get;set;}

    // relationsips
    public User? User{get;set;}
}

File: frontend/src/@types/project.ts
import type { Task } from "./task";
import type { User } from "./user";

export interface Project{
    id:string | number;
    userId:string|number;
    projectTitle:string;
    description:string;
    startDate:Date;
    endDate:Date;
    user?:User;
    tasks:Task[];
}